docs/feature-variables-and-me.md

```diff
@@ -0,0 +1,266 @@
+
+> ⛅️🔬🔭 Nomenclature
+>
+> Much of the literature around the methodology of experiments Nimbus implements has its roots in medical testing. The feature variables API do not require understanding of double blind experiments or data-science, but this document will occasionally use words like "treatment" or "exposure".
+
+## Introdution
+
+The "Feature" in the "Feature Variables API" refers to features of the application. It's pretty abstract, and how the application is divided up into features is up to the product teams. Over time, a feature may be involved in many experiments.
+
+One easy way to start thinking about features, would be to identify user-visible surfaces of the app: the `new-tab` screen, the `app-menu`, the `context-menu`, the `onboarding`.
+
+However, there is one rule:
+
+> ⛅️🔬🔭 Concept
+>
+> *For a given user each feature can only be involved in one experiment at a time*.
+>
+> If we see user change behavior after being exposed to an experimental treatment we need to be able to attribute it to that treatment, not another from a different experiment.
+
+Imagine you're a designer, doing a re-design of the app's menu. It would be natural to call the app menu a "feature" of the app.
+
+You've got some hypotheses around the icons, and whether they should be to right or left of the text. There is also some uncertainty around the copy for each menu item.
+
+The uncertainties and hypotheses you have might translate into variations and variables you might configure the menu with. If these variables and variations are documented, they should travel as an adjunct to or part of the design itself. Later these will be turned into a more formal document that lives with the code, but it is at this stage when they should be thought about.
+
+To narrow the scope for documentation purposes, we'll focus on a small number of variables. Within the `app-menu` feature, we'll consider the menu being made up of menu items, and we'll zoom in on the settings menu item.
+
+As a team communication tool, it may help to consider a JSON object to enumerate the variables that are configurable for the settings icon, and their defaults.
+
+```json
+{
+    "settings-menu-item-title": "Settings",
+    "settings-menu-item-icon": "ic_settings",
+    "settings-menu-item-enabled": true,
+    "settings-menu-item-action": "firefox://settings"
+}
+```
+
+This JSON object looks like what experimenters will be putting into branch configuration screens in Experimenter, under Feature Configuration.
+
+Where did these keys come from? This is not up to Nimbus, but up to the app, i.e. the app team. In this hypothetical case, you have some theories about the title and the icon, and now the app needs to get those values from nimbus.
```

Comment:
By "the app team" do you mean the engineers who will be ultimately implementing the design (in this example)? Or would "team" in this context still mean _anyone_ on the team?

Reply:
(I don't think this sentence necessarily has to change, I'm just curious)

Reply:
Where do the variables come from? Who names them? They're part of the design, and then become part of the implementation. 

My intention was to convey that Nimbus does not take a view on how you arrange the JSON or spell the keys.